### ISSUE TITLE
Remove Docker setup

### ISSUE THREAD

**Issue by cam-schultz** [ava-labs/teleporter#376]
**Context and scope**
[avalanche-cli](https://github.com/ava-labs/avalanche-cli) supports deploying `TeleporterMessenger`, `TeleporterRegistry`, and `awm-relayer` instances for local networks. We should leverage those features for the open-ended development environment included in this repo. It may make sense to remove the Docker environment entirely in favor of running `avalanche-cli`'s local network on native hardware. 

**Discussion and alternatives**
[avalanche-starter-kit](https://github.com/ava-labs/avalanche-starter-kit) provides an all-in-one Teleporter development environment with local workflows as well as remote workflows via Github codespaces. We should consider removing the development environment in `teleporter` altogether in favor of `avalanche-starter-kit`.

**Open questions**
How customizable is the `awm-relayer` deployment via avalanche-cli? We should consider the utility of the avalanche-cli setup for system level development tasks that simultaneously modify `awm-relayer` and `teleporter`. For such tasks, the dev container workflow in `avalanche-starter-kit` may be preferred.

**Comment by feuGeneA** [comment]
I took at look at this.

From what I can tell, for `avalanche-cli`, the Teleporter and AWM Relayer versions that are used must be officially released versions:
- The Messenger contract deployment functionality simply takes the "Deployment_Transaction" asset from a release (eg [v1.0.0](https://github.com/ava-labs/teleporter/releases/tag/v1.0.0)) and submits it (see [here](https://github.com/ava-labs/avalanche-cli/blob/1396808829288bffbfe510dcf0c50d7990ed3d99/pkg/teleporter/teleporter.go#L227)).
- And, the obtaining of the AWM Relayer is also by reference to a `...github.../releases/` url, [here](https://github.com/ava-labs/avalanche-cli/blob/1396808829288bffbfe510dcf0c50d7990ed3d99/pkg/teleporter/relayer.go#L260).

As for `teleporter-starter-kit`, it seems to be well set up for developing applications _on top of_ Teleporter, but does not seem amenable to development of Teleporter itself. This is because it depends on `avalanche-cli` to do the Messenger deployment and to start the AWM Relayer.

In short, at this time, I don't think either of these tools can replace the Docker setup this repo uses for developing the Teleporter contracts.

**Comment by feuGeneA** [comment]
Discussed on a call and decided we should just remove the docker setup entirely since neither CI nor anyone on the team uses that setup.
